src/Microsoft.Docs.LearnValidation/LearnValidationLogger.cs

```diff
@@ -32,9 +57,9 @@ public bool FileHasError(string file)
             }
         }
 
-        public void Log(LearnErrorLevel errorLevel, LearnErrorCode errorCode, string message = "", string file = null)
+        public void Log(LearnErrorLevel errorLevel, LearnErrorCode errorCode, string file, params object[] message)
         {
-            _writeLog?.Invoke(new LearnLogItem(errorLevel, errorCode, message, file));
+            _writeLog?.Invoke(new LearnLogItem(errorLevel, errorCode, FormatMessage(errorCode, message), file));
```

Comment:
nit: so `_writeLog` is marked non-nullable, then can ignore this null check?
```suggestion
            _writeLog(new LearnLogItem(errorLevel, errorCode, FormatMessage(errorCode, message), file));
```